### ISSUE TITLE
Not working in parallel with many processors

### ISSUE THREAD

**Issue by sbna** [RemoteConnectionManager/feature-catalystWrite#3]
I tried running with 8 processors with the scotch decomposition method. Without the live visualization everything seemed to work. When I ran the live visualization I got the following error on 3 different processors at each time step:

ERROR: In /home/grahmatt/OpenFOAM/ThirdParty-v1706/ParaView-v5.4.0/VTK/Common/ExecutionModel/vtkDemandDrivenPipeline.cxx, line 810
vtkCompositeDataPipeline (0xa0267e0): Input for connection index 7 on input port index 0 for algorithm vtkAppendPolyData(0xa00e380) is of type vtkUnstructuredGrid, but a vtkPolyData is required.

The processors that received the errors were random. It seems that there were issues compiling the grid data for the catalyst live visualization. Attached is a screenshot from paraview during the live simulation where I was unable to display the slice in the left window and only got 5 of the 8 processors data in the right window.

**Comment by sbna** [comment]
I separated the domain into 8 equal blocks across each processor, but still got the same results as shown below with the same issues of 3 missing processors.

**Comment by sbna** [comment]
Hmm, not sure what's going on. Could you try running your setup with the attached gridwriter.py script and share the output? It should be a filename_....vtm and a subdirectory filename_.... Looking at that may give me an idea what's wrong.

**Comment by sbna** [comment]
I was able to run a live visualization with 8 processors using the gridwriter.py script without any issues with the live visualization. The file outputs are filename_Index#_proc#.vtu and filename_Index#.pvtu

**Comment by acbauer** [comment]
There is an inconsistency here between what you think is getting created (a multiblock dataset) and what is really getting created (an unstructured grid dataset). @sbna is correct in that there should be a `filename_....vtm ` here if a multiblock dataset is really getting generated in the adaptor.

**Comment by acbauer** [comment]
Yep, `adaptor.C` is what's creating the VTK grid object which is a `vtkUnstructuredGrid`.

**Comment by grahmatt** [comment]
@acbauer can a vtkMultiblock = vtkUnstructuredGrid but a vtkUnstructuredGrid not always equal vtkMultiblock? If that's the case should we always have the adaptor create a vtkMultiblock grid? Are there any issues with a vtkMultiblock that we should consider?

**Comment by sbna** [comment]
@grahmatt I tested it with 8 processors and with the simple partitioner and it works fine with my script. I'll try to test it with the scotch partitioner

**Comment by grahmatt** [comment]
Did you view both the slice and the input grid with the live catalyst adaptor at the same time on separate render windows? I can view the live visualizations of the slice and the input grid separately with no problems, but as soon as I try to view them at the same time I get the error messages and the slice stops showing while the input grid only shows a select number of blocks.

**Comment by sbna** [comment]
No, I viewed them separately. It looks like we used catalyst erroneously or it can be a bug.

**Comment by olesenm** [comment]
This could be related to the patches (not currently used, but still indirectly involved via the vtkPVFoam::convertVolFields method).

With cell to point interpolation, we also build patch interpolators. However, since this chuck of the code was copied from the pv plugin (serial only), we are still doing something with the processor patches instead of stopping before reaching them.
This may not be the problem, but could be another thing to examine.
